kernel/src/table_properties/deserialize.rs

```diff
@@ -0,0 +1,342 @@
+//! For now we just use simple functions to deserialize table properties from strings. This allows
+//! us to relatively simply implement the functionality described in the protocol and expose
+//! 'simple' types to the user in the [`TableProperties`] struct. E.g. we can expose a `bool`
+//! directly instead of a `BoolConfig` type that we implement `Deserialize` for.
+use std::time::Duration;
+
+use super::*;
+use crate::expressions::ColumnName;
+use crate::table_features::ColumnMappingMode;
+use crate::utils::require;
+
+const SECONDS_PER_MINUTE: u64 = 60;
+const SECONDS_PER_HOUR: u64 = 60 * SECONDS_PER_MINUTE;
+const SECONDS_PER_DAY: u64 = 24 * SECONDS_PER_HOUR;
+const SECONDS_PER_WEEK: u64 = 7 * SECONDS_PER_DAY;
+
+impl<K, V, I> From<I> for TableProperties
+where
+    I: IntoIterator<Item = (K, V)>,
+    K: AsRef<str>,
+    V: AsRef<str>,
+{
+    fn from(unparsed: I) -> Self {
+        let mut props = TableProperties::default();
+        for (k, v) in unparsed {
+            let parsed =
+                match k.as_ref() {
+                    "delta.appendOnly" => {
```

Comment:
Ah haha this is funny that I had the same thought and dismissed it as I felt like I was being too performance-concerned! Glad I'm not crazy :)

I think you're correct and the rust compiler doesn't (at least in 1.73 stable) implement anything fancy here - according to quick chatGPT inquiry

really i think our options are:
- leverage custom `Deserialize` from `serde` for constant-time lookup (/linear-time loop).
  - pros: well-known efficient solution, elegant annotations on the struct for e.g. renaming the field for deserialization
  - cons: Extra overhead from `serde` considering we have such a constrained case of only ever deserializing from a `HashMap<String, String>`, code complexity of custom `Deserialize`
- leverage `HashMap`-based lookup instead of the `match` for constant-time.
  - pros: relatively well-known efficient solution, straightforward/understandable
  - cons: increasing code size/complexity increase (similar to serde approach above)
  - extra notes: could even use a crate like `phf` to get compile-time generation of perfect hash functions for string matching.
- leave it as-is 
  - pros: simplest/fastest
  - cons: worst performance

given these options I propose we move forward as-is since this isn't really a blocker but we open an issue to implement option (1) or (2) in the future - this would be a great 'first-issue' for someone to tackle and ramp up on a little piece of the project.

Reply:
https://github.com/delta-io/delta-kernel-rs/issues/535